src/lib/api/basyx-v3/api.ts

```diff
@@ -236,6 +270,52 @@ export const AssetAdministrationShellRepositoryApiFp = function (configuration?:
             };
         },
 
+        /**
+         * @summary Uploads a thumbnail to the specified Asset Administration Shell (AAS).
+         * @param {string} aasId - The unique identifier of the Asset Administration Shell.
+         * @param {Blob} image - The image file to be uploaded as the thumbnail.
+         * @param fileName - Name of the image file to be uploaded.
+         * @param {object} [options] - Optional. Override HTTP request options.
+         * @throws {RequiredError}
+         */
+        putThumbnailToShell(
+            aasId: string,
+            image: Blob,
+            fileName: string,
+            options?: any,
+        ): (fetch?: FetchAPI, basePath?: string) => Promise<Response> {
+            return async (requestHandler: FetchAPI = isomorphicFetch, basePath: string = BASE_PATH) => {
+                const localVarRequestOptions = Object.assign({ method: 'PUT' }, options);
```

Comment:
I find the whole function kind of hard to understand, particularly the Request preparation part, but maybe that is okay considering that we just use it as a "magic black bax that does the thing".